### ISSUE TITLE
Opcode length support for SPC700 (specifically .b and .w)

### ISSUE THREAD

**Issue by KungFuFurby** [RPGHacker/asar#83]
I ran into an unusual situation while attempting to port my SPC700 code from TASM to ASAR...

I used the program counter to create a series of zero page variables (I also did one page variables, although I chose to mostly reserve them for the stack), like these (using the SPC registers as an example)...

```
org $0000F0

;SPC700 Registers defined...
SPC_TEST: ;No comment.
skip 1
SPC_CONTROL: ;Timer and ROM control
skip 1
SPC_DSPADDR: ;DSP Register ID
skip 1
SPC_DSPDATA: ;DSP Register Content
skip 1
SPC_CPUIO0: ;SPC<->SNES data byte 1
skip 1
SPC_CPUIO1: ;SPC<->SNES data byte 2
skip 1
SPC_CPUIO2: ;SPC<->SNES data byte 3
skip 1
SPC_CPUIO3: ;SPC<->SNES data byte 4
skip 1
SPC_AUXIO4: ;Free memory?
skip 1
SPC_AUXIO5: ;Free memory?
skip 1
SPC_T0DIV: ;8010hz timer divider
skip 1
SPC_T1DIV: ;8010hz timer divider
skip 1
SPC_T2DIV: ;64080hz timer divider
skip 1
SPC_T0OUT: ;8010hz timer output
skip 1
SPC_T1OUT: ;8010hz timer output
skip 1
SPC_T2OUT: ;64080hz timer output
skip 1
```

Then I attempted to write something to one of these registers via the accumulator, expecting two bytes instead of three. For example, this snippet of code (I have replaced one of my defines accordingly...

```
snd_clear_dsp:
	clrp					;Zero direct page flag.
	mov A, #$FF
	mov SPC_DSPADDR, #$5C			;Stop all channels first.
	mov SPC_DSPDATA, A			;KOFF will be cleared
	mov Y, #$7F				;later on, though.
	mov A, #$00
```

When I compiled the code, I expected...
$8F $5C $F2 $C4 $F3
Instead, I got...
$8F $5C $F2 $C5 $F3 $00

There is an ambiguity in the syntax for testing the SPC700 that means I have absolutely no clue how to force a label like this to go from $C5 $F3 $00 to $C4 $F3 (as in, force an 8-bit pointer without using DB as an impromptu replacement for a label). I would have a similar situation if I were to use setp and then briefly use the one page section...

This is what I resort to when I find these kind of bugs and I have no idea how to force the issue...

```
snd_clear_dsp:
	clrp
	mov A, #$FF
	mov SPC_DSPADDR, #$5C	
	db $C4					;mov dp, A opcode
	db SPC_DSPDATA
	mov Y, #$7F
	mov A, #$00
```

**Comment by randomdude999** [comment]
This is a workaround, albeit a way better workaround than `db`: Just use defines.
```asm
; SPC700 registers
!SPC_TEST = $F0 ;No comment.
!SPC_CONTROL = $F1 ;Timer and ROM control
!SPC_DSPADDR = $F2 ;DSP Register ID
!SPC_DSPDATA = $F3 ;DSP Register Content
!SPC_CPUIO0 = $F4 ;SPC<->SNES data byte 1
!SPC_CPUIO1 = $F5 ;SPC<->SNES data byte 2
!SPC_CPUIO2 = $F6 ;SPC<->SNES data byte 3
!SPC_CPUIO3 = $F7 ;SPC<->SNES data byte 4
!SPC_AUXIO4 = $F8 ;Free memory?
!SPC_AUXIO5 = $F9 ;Free memory?
!SPC_T0DIV = $FA ;8010hz timer divider
!SPC_T1DIV = $FB ;8010hz timer divider
!SPC_T2DIV = $FC ;64080hz timer divider
!SPC_T0OUT = $FD ;8010hz timer output
!SPC_T1OUT = $FE ;8010hz timer output
!SPC_T2OUT = $FF ;64080hz timer output
```
I haven't really hacked the SPC700 myself, but it looks like the size specification syntax of the 65c816 (appending `.b` or `.w` to the opcode name for a byte or word value) couldn't be used here.

**Comment by KungFuFurby** [comment]
Confirmed. I got an unknown command error (I actually completely missed that in the manual, but it does mean it's still unsupported). However, I like that idea for supporting this...

```
snd_clear_dsp:
	clrp					;Zero direct page flag.
	mov A, #$FF
	mov.b SPC_DSPADDR, #$5C			;Stop all channels first.
	mov.b SPC_DSPDATA, A			;KOFF will be cleared
	mov Y, #$7F				;later on, though.
	mov A, #$00
```

I actually prefer not to use defines to avoid accidentally defining more than one variable.

Interestingly enough, for hacking SPC files (and not creating them from scratch), I used a hex editor. And my kind of hacking would actually be perfect for this assembler... on .spc files, not SNES ROMs. This is because I have to take data straight out of the SNES ROM and implant them into the SPC700 file, then create some custom code to replace what the SNES was doing to the SPC700.

And that's because I'm doing the impossible (well, unless there are memory constraints and there are not enough sound effect samples to overwrite and there's not enough free space elsewhere): taking games that stream their music and playing them back on a .spc file.
